### PR TITLE
fix(compiler): Incorrect focus on error [LNG-274]

### PR DESCRIPTION
**Pull request by DieMyst**
## Description
```
func arr(strs: []string) -> []string
  <- strs
```
this code throws `StringOutOfBound` instead of parser error 

## Proposed Changes
Pad strings that the compiler error renderer can point.

## Checklist
- [x] Corresponding issue has been created and linked in PR title.
- [x] Proposed changes are covered by tests.
- [x] Documentation has been updated to reflect the changes (if applicable).
- [x] I have self-reviewed my code and ensured its quality.
- [x] I have added/updated necessary comments to aid understanding.

## Reviewer Checklist
- [ ] Tests have been reviewed and are sufficient to validate the changes.
- [ ] Documentation has been reviewed and is up to date.
- [ ] Any questions or concerns have been addressed.

### CODE REVIEW

**Comment by DieMyst** [comment]
also fixes LNG-272

File: io/src/test/scala/aqua/RenderingSpec.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package aqua
+
+import aqua.Rendering.given_Show_AquaError
```
Comment (InversionSpaces) [inline io/src/test/scala/aqua/RenderingSpec.scala:3]:
```suggestion
import aqua.Rendering.given
```

File: io/src/test/scala/aqua/RenderingSpec.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+
+    // `.show` don't work for some reason
+    val showResult = given_Show_AquaError.show(
+      AquaError.ParserError[FileModuleId, AquaFileError, FileSpan.F](
+        LexerError[FileSpan.F]((fileSpan, error))
+      )
+    )
+
+    showResult should include("Syntax error: file:1:37")
```
Comment (InversionSpaces) [inline io/src/test/scala/aqua/RenderingSpec.scala:36]:
```suggestion
    val result: AquaError[FileModuleId, AquaFileError, FileSpan.F] =
      AquaError.ParserError(LexerError((fileSpan, error)))

    result.show should include("Syntax error: file:1:37")
```

File: parser/src/main/scala/aqua/parser/Parser.scala [hunk 3]
```diff
@@ -1,21 +1,16 @@
...
-import aqua.parser.Ast.Tree
-import aqua.parser.{Ast, LexerError, ParserError}
 import aqua.parser.expr.RootExpr
 import aqua.parser.head.HeadExpr
-import aqua.parser.lexer.Token
 import aqua.parser.lift.LiftParser.LiftErrorOps
-import aqua.parser.lift.{FileSpan, LiftParser, Span}
+import aqua.parser.lift.Span.S
+import aqua.parser.lift.{LiftParser, Span}
 import cats.data.{Validated, ValidatedNec}
```
Comment (InversionSpaces) [inline parser/src/main/scala/aqua/parser/Parser.scala:8]:
```suggestion

import cats.data.{Validated, ValidatedNec}
```

File: parser/src/main/scala/aqua/parser/lift/Span.scala [hunk 4]
```diff
@@ -51,11 +51,17 @@ object Span {
...
+    val end = idx + len
+    // if we want to point to the place after a string, pad this string with spaces
+    val padded =
+      if (str.length < idx + len) str.padTo(end, ' ')
+      else str
+    FocusedLine(
+      padded.substring(0, idx),
+      padded.substring(idx, end),
+      padded.substring(end)
+    )
```
Comment (InversionSpaces) [inline parser/src/main/scala/aqua/parser/lift/Span.scala:64]: Why not just use `String#slice` method?

File: parser/src/main/scala/aqua/parser/lift/Span.scala [hunk 5]
```diff
@@ -51,11 +51,14 @@ object Span {
...
     idx: Int,
     len: Int
-  ): FocusedLine = FocusedLine(
-    str.substring(0, idx),
-    str.substring(idx, idx + len),
-    str.substring(idx + len)
-  )
+  ): FocusedLine =
+    val end = idx + len
+    // if we want to point to the place after a string, pad this string with spaces
```
Comment (InversionSpaces) [inline parser/src/main/scala/aqua/parser/lift/Span.scala:56]:
```suggestion
```

File: parser/src/main/scala/aqua/parser/lift/Span.scala [hunk 6]
```diff
@@ -51,11 +51,14 @@ object Span {
...
+    FocusedLine(
+      str.slice(0, idx),
+      str.slice(idx, end),
+      str.slice(0, end)
```
Comment (InversionSpaces) [inline parser/src/main/scala/aqua/parser/lift/Span.scala:60]:
```suggestion
      str.slice(end, str.length)
```

File: parser/src/main/scala/aqua/parser/lift/Span.scala [hunk 7]
```diff
@@ -51,11 +51,14 @@ object Span {
     str: String,
     idx: Int,
     len: Int
-  ): FocusedLine = FocusedLine(
-    str.substring(0, idx),
-    str.substring(idx, idx + len),
-    str.substring(idx + len)
-  )
+  ): FocusedLine =
+    val end = idx + len
```
Comment (InversionSpaces) [inline parser/src/main/scala/aqua/parser/lift/Span.scala:55]: nit: Could you wrap the method in `{}`?
I don't like indentation as scope marking

Comment (DieMyst) [reply parser/src/main/scala/aqua/parser/lift/Span.scala:54]: me too

**Review by InversionSpaces — APPROVED** [review]

**Pull request merged**